hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/addressbook/AddressBookServiceImpl.java

```diff
@@ -269,4 +303,92 @@ private void updatePreviousAddressBook(FileData fileData) {
                     }
                 });
     }
+
+    /**
+     * Check whether address book migration should be run based on if flag has been set and if table is empty
+     *
+     * @return Whether to run address book migration
+     */
+    private boolean shouldMigrate() {
+        if (hasAddressBooks) {
+            return false;
+        }
+
+        hasAddressBooks = addressBookRepository.count() > 0;
+        return !hasAddressBooks;
+    }
+
+    /**
+     * Retrieve the initial address book file for the network from either the file system or class path
+     *
+     * @return Address book fileData object
+     */
+    private synchronized FileData getInitialAddressBookFileData() {
+        byte[] addressBookBytes;
+
+        // retrieve bootstrap address book from filesystem or classpath
+        try {
+            Path initialAddressBook = mirrorProperties.getInitialAddressBook();
+            if (initialAddressBook != null) {
+                addressBookBytes = Files.readAllBytes(initialAddressBook);
+                log.info("Loading bootstrap address book of {} B from {}", addressBookBytes.length,
+                        initialAddressBook);
+            } else {
+                MirrorProperties.HederaNetwork hederaNetwork = mirrorProperties.getNetwork();
+                String resourcePath = String.format("/addressbook/%s", hederaNetwork.name().toLowerCase());
+                Resource resource = new ClassPathResource(resourcePath, getClass());
+                addressBookBytes = IOUtils.toByteArray(resource.getInputStream());
+                log.info("Loading bootstrap address book of {} B from {}", addressBookBytes.length, resource);
+            }
+        } catch (Exception e) {
+            throw new IllegalStateException("Unable to load bootstrap address book", e);
+        }
+
+        return new FileData(0L, addressBookBytes,
+                AddressBookServiceImpl.ADDRESS_BOOK_102_ENTITY_ID,
+                TransactionTypeEnum.FILECREATE.getProtoId());
+    }
+
+    /**
+     * Batch parse all 101 and 102 addressBook fileData objects and update the address_book table
+     */
+    private void parseHistoricAddressBooks() {
+        AtomicLong fileDataEntries = new AtomicLong(0);
+        AtomicLong currentConsensusTimestamp = new AtomicLong(0);
+
+        // starting from consensusTimeStamp = 0 retrieve pages of fileData entries for historic address books
+        int pageSize = 1000; //
```

Comment:
Nit, unless there's supposed to be a comment here:
```suggestion
        int pageSize = 1000;
```

Reply:
Will remove thanks

Reply:
Updated